tests/test_module_hamevo.py

```diff
@@ -167,19 +169,56 @@ def test_hamevo_consistency(get_hamiltonians: Callable) -> None:
         ("exp", pyq.HamEvoExp),
     ],
 )
+@pytest.mark.parametrize(
+    "H, t_evo, target, batch_size",
+    [
+        (  # batchsize 1 | 1
+            Hamiltonian(1),
+            torch.tensor([torch.pi / 4], dtype=torch.cdouble),
+            torch.tensor([0.5]),
+            1,
+        ),
+        (  # batchsize 1 | 4
+            Hamiltonian(1),
+            torch.tensor([torch.pi / 4, 0.0, torch.pi / 2, torch.pi]),
+            torch.tensor([0.5, 1.0, 0.0, 1.0]),
+            4,
+        ),
+        (  # batchsize 2 | 1
+            Hamiltonian(2),
+            torch.tensor([torch.pi / 4], dtype=torch.cdouble),
+            torch.tensor([0.5, 0.5]),
+            2,
+        ),
+        (  # batchsize 2 | 2
+            Hamiltonian(2),
+            torch.tensor([torch.pi / 4, torch.pi]),
+            torch.tensor([0.5, 1.0]),
+            2,
+        ),
+    ],
+)
```

Comment:
I don't really like the usage of `Hamiltonian` here for this test just because it's a very strict generator (being diagonal) which could potentially mask incorrect results for other more general cases, but of course I understand why you used it here to compare with the analytical results for specific times. Also since the other tests are also passing with the more general hamiltonian, even if they don't use time batching, I'm ok with leaving it as is.

I feel like the HamEvo code could use a little cleaning anyway and I would be happy to take care of that at some point 😅.

Reply:
yeah, I very much agree. I thought we were going to unify the three evolution methods with #53, but that just resulted in adding another class. I'll write an issue for this and also mention to write better tests